kclvm/tools/src/query/override.rs

```diff
@@ -138,44 +239,192 @@ impl<'ctx> MutSelfMutWalker<'ctx> for OverrideTransformer {
+    }
 }
 
 impl OverrideTransformer {
-    pub(crate) fn _get_schema_config_field_paths(
-        &mut self,
-        schema_expr: &mut ast::SchemaExpr,
-    ) -> (Vec<String>, Vec<String>) {
+    /// Get all field paths from AST nodes including schema and config.
+    #[inline]
+    fn get_field_paths(&mut self, expr: &mut ast::NodeRef<ast::Expr>) -> Vec<String> {
+        match &mut expr.node {
+            ast::Expr::Schema(schema_expr) => self.get_schema_config_field_paths(schema_expr),
+            ast::Expr::Config(config_expr) => self.get_config_field_paths(config_expr),
+            _ => vec![],
+        }
+    }
+
+    /// Get all field paths from a schema AST node.
+    fn get_schema_config_field_paths(&mut self, schema_expr: &mut ast::SchemaExpr) -> Vec<String> {
         if let ast::Expr::Config(config_expr) = &mut schema_expr.config.node {
-            self._get_config_field_paths(config_expr)
+            self.get_config_field_paths(config_expr)
         } else {
-            (vec![], vec![])
+            vec![]
         }
     }
-    pub(crate) fn _get_config_field_paths(
-        &mut self,
-        config: &mut ast::ConfigExpr,
-    ) -> (Vec<String>, Vec<String>) {
+
+    /// Get all field paths from a config AST node.
+    fn get_config_field_paths(&mut self, config: &mut ast::ConfigExpr) -> Vec<String> {
         let mut paths = vec![];
-        let mut paths_with_id = vec![];
         for entry in config.items.iter_mut() {
-            let (mut _paths, mut _paths_with_id) = self._get_key_value_paths(&mut entry.node);
-            paths.append(&mut _paths);
-            paths_with_id.append(&mut &mut _paths_with_id);
+            let mut entry_paths = self.get_entry_paths(&mut entry.node);
+            paths.append(&mut entry_paths);
         }
-        (paths, paths_with_id)
+        paths
     }
-    pub(crate) fn _get_key_value_paths(
-        &mut self,
-        _entry: &mut ast::ConfigEntry,
-    ) -> (Vec<String>, Vec<String>) {
-        (vec![], vec![])
+
+    /// Get all field paths from a config entry.
+    fn get_entry_paths(&mut self, entry: &mut ast::ConfigEntry) -> Vec<String> {
+        let mut paths = vec![];
+        let path = self.get_path_from_key(&entry.key);
+        if !path.is_empty() {
+            paths.push(path.clone());
+            let value_paths = self.get_field_paths(&mut entry.value);
+            if !value_paths.is_empty() {
+                paths.append(
+                    &mut value_paths
+                        .iter()
+                        .map(|value_path| format!("{}.{}", path, value_path))
+                        .collect::<Vec<String>>(),
+                );
+            }
+        }
+        paths
     }
-    pub(crate) fn _find_schema_config_and_repalce(
+
+    /// Get config key path from the AST key node.
+    #[inline]
+    fn get_path_from_key(&mut self, key: &Option<ast::NodeRef<ast::Expr>>) -> String {
+        match key {
```

Comment:
Seems like a mapping func from string-based ast node to a Rust string?

Reply:
> Seems like a mapping func from string-based ast node to a Rust string?

Have moved it into the `kclvm_ast` crate.